source/OutfitterPanel.cpp

```diff
@@ -330,427 +362,573 @@ double OutfitterPanel::DrawDetails(const Point &center)
 	}
+
+	return can_move;
 }
 
 
 
-void OutfitterPanel::FailSell(bool toStorage) const
+bool OutfitterPanel::ButtonActive(char key, bool shipRelatedOnly)
 {
-	const string &verb = toStorage ? "uninstall" : "sell";
-	if(!planet || !selectedOutfit)
-		return;
-	else if(selectedOutfit->Get("map"))
-		GetUI()->Push(new Dialog("You cannot " + verb + " maps. Once you buy one, it is yours permanently."));
-	else if(HasLicense(selectedOutfit->TrueName()))
-		GetUI()->Push(new Dialog("You cannot " + verb + " licenses. Once you obtain one, it is yours permanently."));
-	else
-	{
-		bool hasOutfit = player.Cargo().Get(selectedOutfit);
-		hasOutfit = hasOutfit || (!toStorage && player.Storage().Get(selectedOutfit));
-		for(const Ship *ship : playerShips)
-			if(ship->OutfitCount(selectedOutfit))
-			{
-				hasOutfit = true;
-				break;
-			}
-		if(!hasOutfit)
-			GetUI()->Push(new Dialog("You do not have any of these outfits to " + verb + "."));
-		else
-		{
-			for(const Ship *ship : playerShips)
-				for(const pair<const char *, double> &it : selectedOutfit->Attributes())
-					if(ship->Attributes().Get(it.first) < it.second)
-					{
-						for(const auto &sit : ship->Outfits())
-							if(sit.first->Get(it.first) < 0.)
-							{
-								GetUI()->Push(new Dialog("You cannot " + verb + " this outfit, "
-									"because that would cause your ship's \"" + it.first +
-									"\" value to be reduced to less than zero. "
-									"To " + verb + " this outfit, you must " + verb + " the " +
-									sit.first->DisplayName() + " outfit first."));
-								return;
-							}
-						GetUI()->Push(new Dialog("You cannot " + verb + " this outfit, "
-							"because that would cause your ship's \"" + it.first +
-							"\" value to be reduced to less than zero."));
-						return;
-					}
-			GetUI()->Push(new Dialog("You cannot " + verb + " this outfit, "
-				"because something else in your ship depends on it."));
-		}
-	}
+	if(key == 'b')
+		return static_cast<bool>(CanMoveOutfit(OutfitLocation::Shop, OutfitLocation::Ship));
+	if(key == 'i')
+		return CanMoveOutfit(OutfitLocation::Cargo, OutfitLocation::Ship) ||
+			CanMoveOutfit(OutfitLocation::Storage, OutfitLocation::Ship);
+	if(key == 'c')
+		return !shipRelatedOnly && (CanMoveOutfit(OutfitLocation::Storage, OutfitLocation::Cargo) ||
+			CanMoveOutfit(OutfitLocation::Shop, OutfitLocation::Cargo));
+	if(key == 's')
+		return (!shipRelatedOnly && (CanMoveOutfit(OutfitLocation::Cargo, OutfitLocation::Shop) ||
+			CanMoveOutfit(OutfitLocation::Storage, OutfitLocation::Shop))) ||
+			CanMoveOutfit(OutfitLocation::Ship, OutfitLocation::Shop);
+	if(key == 'u')
+		return CanMoveOutfit(OutfitLocation::Ship, OutfitLocation::Storage) ||
+			(!shipRelatedOnly && CanMoveOutfit(OutfitLocation::Cargo, OutfitLocation::Storage));
+	if(key == 'r')
+		return (!shipRelatedOnly && CanMoveOutfit(OutfitLocation::Cargo, OutfitLocation::Storage)) ||
+			CanMoveOutfit(OutfitLocation::Ship, OutfitLocation::Storage);
```

Comment:
Aren't these the same?
```suggestion
	if(key == 'u' || key == 'r')
		return CanMoveOutfit(OutfitLocation::Ship, OutfitLocation::Storage) ||
			(!shipRelatedOnly && CanMoveOutfit(OutfitLocation::Cargo, OutfitLocation::Storage));
```